### PR TITLE
Add missing query for next sequence send

### PR DESCRIPTION
**Pull request by expertdicer**
## Description

closes: #3383

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/grpc_query_test.go [hunk 1]
```diff
@@ -1565,3 +1565,104 @@ func (suite *KeeperTestSuite) TestQueryNextSequenceReceive() {
...
+				expSeq = 0
+				req = &types.QueryNextSequenceSendRequest{
+					PortId:    path.EndpointA.ChannelConfig.PortID,
+					ChannelId: path.EndpointA.ChannelID,
+				}
+			},
+			true,
+		},
+		{
+			"basic success on ordered channel returns the set receive sequence",
```
Comment (crodriguezvega) [inline modules/core/04-channel/keeper/grpc_query_test.go:1632]:
```suggestion
			"basic success on ordered channel returns the set send sequence",
```

**Review by crodriguezvega — CHANGES_REQUESTED** [review]
Thank you, @expertdicer. There are a few places where `receive` is used instead of `send`.

File: modules/core/04-channel/client/cli/query.go [hunk 2]
```diff
@@ -455,3 +455,38 @@ func GetCmdQueryNextSequenceReceive() *cobra.Command {
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			portID := args[0]
+			channelID := args[1]
+			prove, _ := cmd.Flags().GetBool(flags.FlagProve)
+
+			sequenceRes, err := utils.QueryNextSequenceSend(clientCtx, portID, channelID, prove)
```
[the remark this answers is not in the document]
Comment (expertdicer) [reply modules/core/04-channel/client/cli/query.go:478]: i think ``sequenceRes`` means sequence response

[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/core/04-channel/client/cli/query.go:478]: Yeah, indeed, you're right! :)

File: proto/ibc/core/channel/v1/query.proto [hunk 3]
```diff
@@ -374,3 +380,23 @@ message QueryNextSequenceReceiveResponse {
...
+// QueryNextSequenceSendRequest is the request type for the
+// Query/QueryNextSequenceSendRequest RPC method
+message QueryNextSequenceSendRequest {
+  // port unique identifier
+  string port_id = 1;
+  // channel unique identifier
+  string channel_id = 2;
+}
+
+// QuerySequenceResponse is the request type for the
```
Comment (crodriguezvega) [inline proto/ibc/core/channel/v1/query.proto:393]:
```suggestion
// QueryNextSequenceSendResponse is the request type for the
```

File: proto/ibc/core/channel/v1/query.proto [hunk 4]
```diff
@@ -374,3 +380,23 @@ message QueryNextSequenceReceiveResponse {
...
+// Query/QueryNextSequenceSendResponse RPC method
```
Comment (crodriguezvega) [inline proto/ibc/core/channel/v1/query.proto:394]:
```suggestion
// Query/QueryNextSequenceSend RPC method
```

File: proto/ibc/core/channel/v1/query.proto [hunk 5]
```diff
@@ -374,3 +380,23 @@ message QueryNextSequenceReceiveResponse {
   // height at which the proof was retrieved
   ibc.core.client.v1.Height proof_height = 3 [(gogoproto.nullable) = false];
 }
+
+// QueryNextSequenceSendRequest is the request type for the
+// Query/QueryNextSequenceSendRequest RPC method
```
Comment (crodriguezvega) [inline proto/ibc/core/channel/v1/query.proto:385]:
```suggestion
// Query/QueryNextSequenceSend RPC method
```

**Review by crodriguezvega — APPROVED** [review]
Thanks, @expertdicer. I just realised there were a couple of other naming typos, but apart from that it looks good to me. I think we can ignore the lint warning `receiver name k should be consistent with previous receiver name q for Keeper`, since all the other receiver names will be renamed in a different PR.

**Comment by expertdicer** [comment]
> Thanks, @expertdicer. I just realised there were a couple of other naming typos, but apart from that it looks good to me. I think we can ignore the lint warning `receiver name k should be consistent with previous receiver name q for Keeper`, since all the other receiver names will be renamed in a different PR.

I fixed the typos but I think there are typos in ``QueryNextSequenceReceive`` too. Should I fix it too ?

**Review by damiannolan — APPROVED** [review]
Nice! Thanks for the PR @expertdicer, very clean!

File: modules/core/04-channel/keeper/grpc_query.go [hunk 6]
```diff
@@ -541,6 +541,41 @@ func (q Keeper) NextSequenceReceive(c context.Context, req *types.QueryNextSeque
...
+// NextSequenceSend implements the Query/NextSequenceSend gRPC method
+func (k Keeper) NextSequenceSend(c context.Context, req *types.QueryNextSequenceSendRequest) (*types.QueryNextSequenceSendResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	if err := validategRPCRequest(req.PortId, req.ChannelId); err != nil {
+		return nil, err
+	}
```
Comment (DimitrisJim) [inline modules/core/04-channel/keeper/grpc_query.go:553]: As per #3422, we should move the unwrapping after the error handling, this is similar to how other queries are doing it (i.e `ChannelClientState` and `ChannelConsensusState`):

```suggestion
	if req == nil {
		return nil, status.Error(codes.InvalidArgument, "empty request")
	}

	if err := validategRPCRequest(req.PortId, req.ChannelId); err != nil {
		return nil, err
	}
	
	ctx := sdk.UnwrapSDKContext(c)
```

Comment (expertdicer) [reply modules/core/04-channel/keeper/grpc_query.go]: Yes. Thank you

**Comment by crodriguezvega** [comment]
> I fixed the typos but I think there are typos in QueryNextSequenceReceive too. Should I fix it too ?

No, no need to fix those in this PR. :) We can do that in a follow-up.

**Review by DimitrisJim — APPROVED** [review]
lgtm!

**Pull request merged**